pkg/types/types_test.go

```diff
@@ -0,0 +1,473 @@
+/*
+Copyright © 2020 Bob Callaway <bcallawa@redhat.com>
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package types
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"strings"
+	"testing"
+
+	"go.uber.org/goleak"
+)
+
+func TestMain(m *testing.M) {
+	goleak.VerifyTestMain(m)
+}
+
+type BadReader struct {
+}
+
+func (br BadReader) Read(p []byte) (n int, err error) {
+	return 0, errors.New("test error")
+}
+
+func TestRekorLeaf(t *testing.T) {
+
+	type Test struct {
+		caseDesc      string
+		json          string
+		expectSuccess bool
+	}
+
+	tests := []Test{
+		{
+			caseDesc: "Valid full entry",
+			json: `{
+				"URL": "https://mirrors.edge.kernel.org/pub/linux/kernel/v5.x/linux-5.8.9.tar.gz",
```

Comment:
We don't actually hit these URLs, right? I don't think we do here, just want to double check though.

Reply:
correct. I can change this value to something else since the point of this particular test is to cover extra fields in the JSON